src/pepper/evaluator.py

```diff
@@ -0,0 +1,189 @@
+#! /usr/bin/env python3
+
+# This file is a part of the Pepper project, https://github.com/devosoft/Pepper
+# (C) Michigan State University, under the MIT License
+# See LICENSE.txt for more information
+
+"""
+This is the evaluator module for Pepper
+
+This evaluator is specific to parsing and evaluating macro calls within the if directive.
+This should not be used in isolation.
+"""
+
+import pepper.abstract_symbol_tree as ast
+import pepper.symbol_table as symtable
+from typing import Union, List, cast
+from pepper.symbol_table import Node  # NOQA
+
+
+def validate(exp: Union['Node']) -> str:
+    literal = ""
+    if isinstance(exp, ast.PreprocessingNumberNode) or \
```

Comment:
so i changed to from `Union['Node']` to `Node` I tried to change it to Primitive Node, because it gets called technically on any node, unless I add another check before I call validate that it is of type PrimitiveNode which becomes a bit redundant

Reply:
Sorry,  here I meant that isinstance should a PrimitiveNode instead of having `isinstance(ASCIIlit) or isinstance(operator) or isinstance(preprocessingnumber)`

Reply:
so the problem with that is string literal is also a PrimitiveNode but i treat that differently

Reply:
okay, fair enough